### DISCUSSION TITLE
Small update of env.example

### DISCUSSION THREAD

**Discussion by somethingGoneWrong** [firefly-iii/firefly-iii#5512] [General]
At the moment the env.example file in the app_key entry looks like this:
```
# The encryption key for your sessions. Keep this very secure.
# Change it to a string of exactly 32 chars or use something like `php artisan key:generate` to generate it.
# If you use Docker or similar, you can set this variable from a file by using APP_KEY_FILE
APP_KEY=SomeRandomStringOf32CharsExactly
```
In the migration of the app to docker I lost a lot of time because I was using an invalid key, I think this part should be updated in this way:
```
# The encryption key for your sessions. Keep this very secure.
# Change it to a string of exactly 32 chars or use something like `php artisan key:generate` to generate it.
# This key can look like this <45 string chars>
# APP_KEY=<base64:SomeRandomStringOf45CharsExactly>
# If you use Docker or similar, you can set this variable from a file by using APP_KEY_FILE
APP_KEY=SomeRandomStringOf32CharsExactly
```
I think this may make it clearer how to handle this key, what do you think?

**Comment by JC5** [comment]
What have you been running into, exactly?

**Reply by somethingGoneWrong** [reply]
When I try to migrate my old FF3 setup to a new docker one, I have found it difficult to locate the app_key because I think the documentation is unclear. I made several tests and the only way to read the dump of the old db on the new installation is to copy the app_key entry in the new setup equal from the old installation, and this is composed by the string base64:<45 chars> and not the 32 chars indicate in the docs.
Maybe there was a way to get this string of 32 chars from that value [base64:45 Chars]? thx

**Reply by JC5** [reply]
Last things first: if you do a base64 decode you should get the 32 key value: https://www.base64decode.org/

Since you're migrating from a local installation to a Docker one, you can use the values from your old `.env` file in your new Docker installation. Either use the file verbatim or just set them up as individual `-e VALUE=bla` in your Docker command.

At some point, Laravel made it possible to use `base64:abcde` values. So yes, it's possible you have a `base64:` value instead of a "normal" key. This is something you can find online as well, if you [know what to look for](https://www.google.com/search?q=base64+laravel+app+key). I've not had questions yet about this particular migration issue myself.

I will not add the sentence to the `.env.example` file as you suggested, because it'll just make things more confusing. I'm currently not offering a "migrate to Docker" track because very few people do so. The people that have done it in the past, have simply set up the `base64:` key and thought nothing of it. 

This just isn't something that a lot of people run into. I expect that this issue will help them along the way.

**Comment by jimkoen** [comment]
Short question: Why is this value not explicitly mentioned in the docs anymore? There used to be a big deal of changing the App Key to a custom one, as well as changing the default DB user and password.
I.e. why was this removed?

EDIT: Nevermind, It's still there, I'm just blind as a snake.
